res/controllers/Denon-MC7000-scripts.js

```diff
@@ -837,15 +837,24 @@ MC7000.censor = function(channel, control, value, status, group) {
     }
 };
 
+// Auto-Loop Button
+MC7000.autoLoop = function(channel, control, value, status, group) {
+    if (engine.getValue(group, "loop_enabled")) {
+        script.toggleControl(group, "reloop_toggle");
+    } else {
+        script.toggleControl(group, "beatloop_activate");
+    }
+};
+
```

Comment:
Looks reasonable to me. Just wanna check what happens if a loop is set but inactive. Does it trigger a new loop at play position or reloop the existing one... Can do that by Thursday

Reply:
I like that little improvement.

as all `var` were replaced with `let` then the script doesnt work in Mixxx 2.3 any longer. I have therefore quickly changed my one and could push this to #4021 if you like. Or we let this one still move.

Reply:
I couldn't get `beatloop_activate` to work properly if there was no loop set in the track yet. Anyhow `beatloop_4_activate` works to set a new loop and activate it. Is this a bug or what else do we miss here?

Reply:
I found another issue ... Assumed there was a loop inside the track and I reloop it with `Shift` and `Autoloop`, go out the loop by pressing `Autoloop` then the loop position sometimes move to the current play position and activates. This is quite unpredictable - it doesn't happen always but now and then. Expecting the loop to stay where it was but not toggled on to current play position as I did only wanted to exit the `Reloop`.

Reply:
I am using a much more complex logic for this purpose:

https://github.com/mixxxdj/mixxx/blob/de1853c1caf4a4b0d891b8052d617890fb88f2e6/res/controllers/Denon-MC6000MK2-scripts.js#L948

Don't ask me why I came up with this, I don't remember.

We need a domain specific API to unify this behavior for multiple controllers. Manipulating individual control values is too low level. Needing to write a custom function `hasLoop()` only to determine if a valid loop range is defined clearly indicates this conceptual gap.

Reply:
@uklotzde - Awesome! Thanks for the hint. To add a check for an existing loop in works to solve and set the initial loop if no one was set yet.

```
MC7000.autoLoop = function(channel, control, value, status, group) {
    if (engine.getValue(group, "loop_in") || engine.getValue(group, "loop_enabled")) {
        script.toggleControl(group, "reloop_toggle", 100);
    } else {
        script.toggleControl(group, "beatloop_activate", 100);
    }
};
```
To leave reloop without to set a new activated loop needs to be fixed yet.

Reply:
> script.toggleControl(group, "reloop_toggle", 100);

The third parameter (`100`) doesn't do anything.

Reply:
hm...it still does not what I'd expect.
I think I have to read the API docs more accurate to understand the difference behaviors between
`loop_*`, `reloop_*` and `beatloop_*`

Reply:
So what would be the workflow that you expect?

Reply:
The method `script.toggleControl` does not work for me.
At least the last commit (c43ac92)  is working at least for 90% of my usecases:
Create and activate a beatloop or disable a loop.

Reply:
That commit seems to have some significant flaws in it (see eslint output). Also make sure you are using the correct Control Objects...

Reply:
> > script.toggleControl(group, "reloop_toggle", 100);
> 
> The third parameter (`100`) doesn't do anything.

Mmhhhh interesting - got this line from the [Mixxx Controls page inside the manual](https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#controlpotmeter-controls). See last example in that linked chapter.